art/estimators/object_detection/pytorch_yolo.py

```diff
@@ -0,0 +1,490 @@
+                    predictions_x1y1x2y2_tensor[i_prediction]["masks"].detach().cpu().numpy().squeeze()
+                )
+
+        return predictions_x1y1x2y2_array
+
+    def fit(self, x: np.ndarray, y, batch_size: int = 128, nb_epochs: int = 20, **kwargs) -> None:
+        raise NotImplementedError
+
+    def get_activations(
+        self, x: np.ndarray, layer: Union[int, str], batch_size: int, framework: bool = False
+    ) -> np.ndarray:
+        raise NotImplementedError
+
+    def compute_losses(
+        self, x: np.ndarray, y: List[Dict[str, Union[np.ndarray, "torch.Tensor"]]]
+    ) -> Dict[str, np.ndarray]:
+        """
+        Compute all loss components.
+
+        :param x: Samples of shape (nb_samples, nb_features) or (nb_samples, nb_pixels_1, nb_pixels_2,
+                  nb_channels) or (nb_samples, nb_channels, nb_pixels_1, nb_pixels_2).
+        :param y: Target values of format `List[Dict[Tensor]]`, one for each input image. The
+                  fields of the Dict are as follows:
+
+                  - boxes (FloatTensor[N, 4]): the predicted boxes in [x1, y1, x2, y2] format, with values \
+                    between 0 and H and 0 and W
+                  - labels (Int64Tensor[N]): the predicted labels for each image
+                  - scores (Tensor[N]): the scores or each prediction.
+        :return: Dictionary of loss components.
+        """
+        output_tensor, _, _ = self._get_losses(x=x, y=y)
+        output = {}
+        for key, value in output_tensor.items():
+            output[key] = value.detach().cpu().numpy()
+        return output
+
+    def compute_loss(  # type: ignore
+        self, x: np.ndarray, y: List[Dict[str, Union[np.ndarray, "torch.Tensor"]]], **kwargs
+    ) -> Union[np.ndarray, "torch.Tensor"]:
+        """
+        Compute the loss of the neural network for samples `x`.
+
+        :param x: Samples of shape (nb_samples, nb_features) or (nb_samples, nb_pixels_1, nb_pixels_2,
+                  nb_channels) or (nb_samples, nb_channels, nb_pixels_1, nb_pixels_2).
+        :param y: Target values of format `List[Dict[Tensor]]`, one for each input image. The
+                  fields of the Dict are as follows:
+
+                  - boxes (FloatTensor[N, 4]): the predicted boxes in [x1, y1, x2, y2] format, with values \
+                    between 0 and H and 0 and W
+                  - labels (Int64Tensor[N]): the predicted labels for each image
+                  - scores (Tensor[N]): the scores or each prediction.
+        :return: Loss.
+        """
+        import torch  # lgtm [py/repeated-import]
+
+        output, _, _ = self._get_losses(x=x, y=y)
+
+        # Compute the gradient and return
+        loss = None
+        for loss_name in self.attack_losses:
+            if loss is None:
+                loss = output[loss_name]
+            else:
+                loss = loss + output[loss_name]
+
+        assert loss is not None
+
+        if isinstance(x, torch.Tensor):
```

Comment:
the type hinting for _get_losses above has torch.Tensor rather than numpy arrays being returned in the tuple. Is this check needed here?

Reply:
The typing for x should include tensors, I have updated the typing. That way we return the same type as the input type.